src/syscall_sdl.c

```diff
@@ -47,38 +41,65 @@ typedef struct {
 } event_t;
 
 typedef struct {
-    event_t events[EVENT_QUEUE_CAPACITY];
-    size_t start, end;
-    bool full;
+    event_t *base;
+    size_t end;
 } event_queue_t;
 
+enum {
+    RELATIVE_MODE_SUBMISSION = 0,
+};
+
+typedef struct {
+    uint32_t type;
+    union {
+        union {
+            uint8_t enabled;
+        } mouse;
+    };
+} submission_t;
+
+typedef struct {
+    submission_t *base;
+    size_t start;
+} submission_queue_t;
+
 static SDL_Window *window = NULL;
 static SDL_Renderer *renderer;
 static SDL_Texture *texture;
+static uint32_t queues_capacity = 0;
+static uint32_t event_count;
 static event_queue_t event_queue = {
-    .events = {},
-    .start = 0,
+    .base = NULL,
     .end = 0,
-    .full = false,
+};
+static submission_queue_t submission_queue = {
+    .base = NULL,
+    .start = 0,
 };
 
-static bool event_pop(event_t *event)
+static submission_t submission_pop(void)
 {
-    if (event_queue.start == event_queue.end)
-        return false;
-    *event = event_queue.events[event_queue.start++];
-    event_queue.start &= EVENT_QUEUE_CAPACITY - 1;
-    event_queue.full = false;
-    return true;
+    submission_t submission = submission_queue.base[submission_queue.start++];
+    if (queues_capacity & 1)
```

Comment:
Can you simply guarantee that queue capacity is always power of 2?

Reply:
Therefore, we can write down efficient and short code.

Reply:
Should we simply require the user to select a size that is a power of 2 and notify them when they don't; or should we just align the size up to a power of 2 while allocating and let the user push and pop with different sizes?

Reply:
I prefer the way to align.